FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataWriter/Azure/AzureBlobContainerClientFactory.cs

```diff
@@ -30,10 +30,10 @@ public class AzureBlobContainerClientFactory : IAzureBlobContainerClientFactory
 
             _credentialProvider = credentialProvider;
             _loggerFactory = loggerFactory;
-            _storageConfiguration = storageConfiguration.Value;
+            _externalConnectionString = storageConfiguration.Value.ExternalStorageConnectionString;
```

Comment:
So we suppose the blob client factory will only create external storage client with connection string?

Reply:
The **connectionString** is just for tests, and if **connectionString** is not provided, will create external storage based on **storageUrl**.